src/pk_core.c

```diff
@@ -1217,6 +1217,14 @@ Var varBitRshift(PKVM* vm, Var v1, Var v2) {
   return VAR_NULL;
 }
 
+Var varBitNot(PKVM* vm, Var v) {
+  int64_t i;
+
+  if (isInteger(v, &i))
+    return VAR_NUM((double)(~i));
+  return VAR_NULL;
```

Comment:
`isInteger(v, &i)` will check if it's an integer. and `~"foo"` will result to `null` where it should be a run time error.

And for now, it only could be an integer. Use `validateInteger` to set error if it's not an integer.

```c
if (!validateInteger(vm, v2, &i, ...)) return VAR_NULL;
return VAR_NUM((double)(~i));
```

(And sorry for missing this the first time I reviewed.)